src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -435,6 +443,19 @@ public TStructuralType CopyChangedValues(TStructuralType original)
                 throw Error.Argument(nameof(original), SRResources.DeltaTypeMismatch, _structuredType, original.GetType());
             }
 
+            //To apply ODataId based handler to apply properties on original.
+            if (apiHandlerFactory != null)
+            {
+                IODataAPIHandler refapiHandler = apiHandlerFactory.GetHandler(ODataPath);
+
+                if (refapiHandler != null && ODataPath.Any() && apiHandler.ToString() != refapiHandler.ToString())
```

Comment:
Curious about `apiHandler.ToString() != refapiHandler.ToString()`... Do derived types of `IODataHandler` implement `ToString` method and what do they return? Why is it relevant to compare the two here?

Reply:
Refactored some logic and removed this code here. I moved it to `DeltaSet<T>` and explained why we do the comparison.